### ISSUE TITLE
Grass too high should slow down the mower

### ISSUE THREAD

**Issue by alexvaut** [ClemensElflein/open_mower_ros#54]
When the grass is too high, the mower continues at the same speed, leaving the grass as is.
Implementation:
First we must allow the user to define the prefered speed and then, optionally, when the mower detects that the mower motor speed is too low then it will decrease the mower wheel speed during 1 minute or so. Everything should be configurable through env var.

**Comment by karlranseierausrom** [comment]
I agree to this point, because I have areas, where it did not cut the grass. My idea (withut knowing how to program it) would be to have the following config:
normal_mow_current=1  //current in Ampere which is accpted to be normal. If mow motor current is above this leven, driving speed will be reduced
reduce_velocity_slope= 25// reduction of speed of mow motor current in Percent Ampere. Meaning if the current is above 1 A, it will reduce the velocity by 25% per A. For example if mow current is 3 A, the new velocity will be 

new veocity=velocity-velocity*( (mow motor current-normal_mow_current )*reduce_velocity_slope/100

in the example: new velocity=velocity-velocity((3A-1A)*25/100)=0.5 velocity.
or if current is 2A:
new velocity = 0.75 velocity.

If current is 6A:
new velocity = - 0.25 velocity. Meaning the mower will drive backwards. This will be finde, because if the mower current is reduced, it will directly go forward again. I think that is good. If there are doubts, one could add a protection, that speed become 0 or another value at minimum.

What could happen: There is something which can not be mowed and increases the mow motor current, it will drive forward/backward/forward/backward... I don't care ;-)

**Comment by karlranseierausrom** [comment]
There was a wrong config on my XESC with nearly no power for the mowing motor. I changed it as described in Discord, now this is not an issue for me anymore. Still long grass is not cut as perfect as shorter grass. Thereore:

I have areas which grow much faster (new grass). At that places I recorded the area twice with slightly different areas. Than it will autoamtically mow there twice with a little offset in the lanes. Longer parts of the grass are than completely cut.

**Comment by jeremysalwen** [comment]
This is handled by @Pete_MI632's speed limiter patch:  https://github.com/jeremysalwen/open_mower_ros/tree/speed_limiter_rebase    (I have been using it for over a year now). However some work needs to be done cleaning it up before it's ready to be merged.
